src/peft/tuners/lora/layer.py

```diff
@@ -56,6 +57,8 @@ def _init_empty_weights(self, cls, *args, **kwargs) -> None:
         self.to_empty(device=final_device)
 
     def update_layer(self, adapter_name, r, lora_alpha, lora_dropout, init_lora_weights):
+        if r <= 0:
```

Comment:
How come we don't allow `r=0` anymore? I agree that it doesn't make sense, but our existing code allows this, so technically this is BC breaking.

Reply:
yeah, I wanted to reduce the conditional logic in forward methods as much as possible and frankly `r=0` doesn't make sense and it was the remnant of the original LoRA implementation. I think this should be fine.

Reply:
I agree, always found it odd but never bothered to ask. This explains it. Not sure if there's any code in the wild with `r=0`. Maybe we should mention this hypothetical breaking change in the next release notes.

Reply:
Okay, yes, let's do that